### PR TITLE
MAINT: Remove deprecation for kind

### PR DESCRIPTION
**Pull request by larsoner**
Post-release deprecation resolution

@drammock there is a remaining one in `mne/viz/topomap.py`, feel free to take care of it by pushing commits here or by opening a separate PR

### CODE REVIEW

**Comment by drammock** [comment]
@larsoner there were a TON of things to remove; thanks for noticing / reminding that I forgot to do this after release.

File: mne/viz/topomap.py [hunk 1]
```diff
@@ -3081,14 +2972,6 @@ def plot_regression_weights(
     %(ch_type_topomap)s
     %(sensors_topomap)s
     %(show_names_topomap)s
-
-        .. versionadded:: 1.2
```
Comment (drammock) [inline mne/viz/topomap.py:3085]: nb: this removal is not a mistake; the whole function was added in 1.2 so no need to point out that this param was also added then.

**Comment by mmagnuski** [comment]
Just BTW - why was it decided to remove `vmin` and `vmax` and use `vlims` for topomaps? As long as I remember I used `vmin` and `vmax` in mne and I don't think I ever encountered `vlims` in topomap plotting.

**Comment by drammock** [comment]
> Just BTW - why was it decided to remove `vmin` and `vmax` and use `vlims` for topomaps?

There are some topomap functions where `vlim='joint'` is an option (to compute vmin/vmax automatically and have it be the same for all subplots).  I think `plot_projs_topomap` was the original use case for this.  Since the goal was to standardize the param names across all topomap-generating funcs/methods, switching them all to `vlim` was the simplest way.

**Comment by mmagnuski** [comment]
Ok, I get it now :)

**Pull request merged**